tests/data/config_test_not_case_sensitive_option.json

```diff
@@ -0,0 +1,13 @@
+{
+    "caseSensitive": false,
```

Comment:
There could be tests for:

* `"caseSensitive": true`
* not defining it at all
* what happens with a mixture of cases in the input data
* what happens with ALL UPPER CASE in the input data
* what happens with alllowercase in the input data

Reply:
I added some tests I will add a PR specifically on tests but I think the caseSensitive is well covered now.

Reply:
In my opinion, to well cover this you need multiple test datasets that explore different. A single test function isn't enough. There at least needs to tests for:

re, caseSensitive
re, not caseSensitive
fnmatch, caseSensitive
fnmatch, not caseSensitive

So that's at least four `def test_...():`s. And within each of those four, there needs to be example datasets that get sorted differently depending on the algorithm's settings, so it's probably at least 8.

If you want to do them in a separate PR that's okay. I would encourage you to make an issue to describe exactly what you're going to cover before you do it because there's a combinatorial number of cases to check.